CMakeLists.txt

```diff
@@ -358,6 +358,14 @@ if(APPLE)
 	endif()
 endif(APPLE)
 
+# BSD
+if(${CMAKE_SYSTEM_NAME} MATCHES "BSD")
+	set(BSD TRUE)
+	if(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
+		set(FREEBSD TRUE)
+	endif()
+endif()
+
```

Comment:
If we support more other system we should move that out to some extra file we can include but for now very nice ;).